### ISSUE TITLE
As a National System Admin, I would like to configure the date of birth (mother, father deceased, informant), to allow just the age to be added

### ISSUE THREAD

**Issue by jpye-finch** [opencrvs/opencrvs-core#4005]
### Description
- the exact date of birth is not always known. 
- some countries allow to just provide the age (yyyy)
- this ticket is to add a config option to date of birth to ask for age if dd/mm/yyyy is not known
- if set to true. then a checkbox appears below d.o.b and on check the user is prompted to add the age of (mother, father deceased, informant).

... or only the YYYY input field is mandatory. 

### Acceptance criteria

GIVEN
WHEN
THEN

### Design
...

### Dev tasks

- [ ] Task 1

**Comment by euanmillar** [comment]
@anny320 essentially @rikukissa has already done this in a fork.  This ticket must remain open but we need to remember this ticket is essentially to make this enabled for core - E.G. other countries apart from Nigeria

**Comment by euanmillar** [comment]
@anny320 @jpye-finch @eduffus  as described in #1804 when performing de-duplication on a birth record we check against the date of birth of mother as a value therefore we need a D.O.B. to be generated in the database by default.  Our approach for Nigeria defaults the mother's date of birth to be the same day and month of the child minus the number of years entered.  I hope that is OK.  @rikukissa can you copy a link to the PR that contains this work so we can review and copy into core?

**Comment by rikukissa** [comment]
Sure can, here's the code: https://github.com/opencrvs/opencrvs-core/pull/4054/commits/3d6d8c9173895d251ab47f22ab4d2533f57843a1

The current implementation is a bit naive as we store the mother's age as a `dd.mm.yyyy` with the year calculated based on the given years and `dd.mm` is the same as the child's birthdate. 

Let me know if you need any help with this

**Comment by euanmillar** [comment]
Just waiting for a design for configuring this from @jpye-finch

**Comment by euanmillar** [comment]
@jpye-finch we need the configuration design.  Also the styling for the input with "years" label, looks like the weight of child with "kg" label.  Are both components to be rebuilt as per this design?

**Comment by jpye-finch** [comment]
added link to config design

**Comment by jpye-finch** [comment]
yes styling updated for prefix and postfix. This could be looked at when we refactor < TextInput /> I assume?

**Comment by jpye-finch** [comment]
In form config:
GIVEN I select the Age of [Mother, Father, Deceased, Informant] field
THEN  'Hide field' toggle = disabled off, 
THEN 'Required for registration' toggle = disabled on

In declaration form:
GIVEN I am completing a declaration form
WHEN 'Exact data of birth unknown' is true
THEN Age of [Mother, Father, Deceased, Informant] is required for registration

**Comment by SyedaAfrida** [comment]
**Bug:**
1. Hide and required for registration should not be available in form configuration.
![a](https://user-images.githubusercontent.com/94350336/215984319-24a44e2f-0821-418c-abef-24d719bf6566.png)

2. No error message showing when age of child is greater than mother/ father

https://user-images.githubusercontent.com/94350336/215984672-89e7a423-cbe8-4557-b22d-5835366bd040.mp4

similarly, no error message showing when Date of death is before Deceased DOB
![death age](https://user-images.githubusercontent.com/94350336/215985223-f6e3cd00-cdcd-46b8-bb93-eb3657a49a9f.png)

**Comment by SyedaAfrida** [comment]
@anny320 this ticket is fixed and can be closed
![1](https://user-images.githubusercontent.com/94350336/222665845-5953cc28-9cdf-4b48-84fc-e7ad0bfd53d8.png)


https://user-images.githubusercontent.com/94350336/222665693-e55b44e2-f696-4bdc-b35c-b8e501ccb691.mp4

**Comment by anny320** [comment]
completed
